### DISCUSSION TITLE
[質問]: 画像をスマホの画像フォルダにアクセスし、表示。その画像をRealmに保存してTableViewに表示したい

### DISCUSSION THREAD

**Discussion by sheep1857** [lit-iphone-course/lit-iphone-course#152] [Question-original]
### エラーの概要

画像をRealmに保存して表示する方法を検索したところ、さまざまな記事が出てきたので読みましたが、やり方が複雑で理解できませんでした。
AddVCで画像を追加し、その情報をRealmに保存→リストで探すVCに画像を表示をしたいです。

### GitHubのURL

https://github.com/sheep1857/GoshuinApp.git

### TODOリスト

- [X] GitHubに最新のコードをアップロードしました。
- [X] できるだけ具体的なタイトルを設定しました。

**Comment by txxxxc** [comment]
> さまざまな記事が出てきたので読みましたが

具体的にどんな記事を参考にしたか具体的な例を教えてもらえる助かります！！その中から「この記事の通りやればできるよ！」というおすすめができればと思っています！！

個人的にはこちらの記事の2番で紹介されている方法ですすめると一番きれいに実装できそうですー！
[参考記事](https://zenn.dev/kazushino/articles/90bafb36f07fb2#2.%E7%94%BB%E5%83%8F%E3%82%92document%E3%83%95%E3%82%A9%E3%83%AB%E3%83%80%E3%81%AB%E4%BF%9D%E5%AD%98%E3%81%97%E3%80%81realm%E3%81%AB%E3%81%AF%E3%81%9D%E3%81%AE%E3%83%95%E3%82%A1%E3%82%A4%E3%83%ABurl%E3%82%92%E4%BF%9D%E5%AD%98%E3%81%99%E3%82%8B)

上記のRealmの記事で出てくるDocument周りに関してはこちらの記事で詳しく解説されてるので、確認してみてください！！

https://tech.amefure.com/swift-file-manager

**Reply by sheep1857** [reply]
サイトに沿ってやってみたところ、tableData の定義付けができず、エラーになってしまいました、、
また、三つ目のエラーについても調べてみましたが、あまりよく理解できず解決することができませんでした

<img width="780" alt="スクリーンショット 2023-03-19 13 50 56" src="https://user-images.githubusercontent.com/105029351/226154397-fb156c40-c6e1-4a7d-938b-36ae7fe9ec6d.png">

**Reply by txxxxc** [reply]
ちなみにバグってるところまでのコードってGitHubに上がってますか...?

多分記事に書いてあってtableDataは記事を書いた人のプロジェクトに書いてある変数なので、自分のプロジェクトで使う場合は読み替える必要があります！
多分しーぷのプロジェクトで言うと、GoshuinListになるんじゃないかなと思います！！

**Reply by sheep1857** [reply]
はい、gitに上がってます！
https://github.com/sheep1857/GoshuinApp.git

tableDataのところ、私もgoshuinListだと思ってやってみたのですが、このようにエラーが出てしまいます、、
<img width="814" alt="スクリーンショット 2023-03-19 14 28 54" src="https://user-images.githubusercontent.com/105029351/226155693-8d13543f-b2f3-407d-acd6-510a80803633.png">

**Comment by appleEmily** [comment]
①「tableDataのところ、私もgoshuinList」でももはやなくて、最新では「RealmData」になってたよ

②エラーについて　goshuinListは
```
// RealmData型の変数を用意（まだ空の配列）
    var goshuinList: [RealmData] = []
```
ってやっているのだけど、右辺に置いているrealm.objects(〜〜)ってやつは、Results型ってやつなので、方が違ってしまっているので代入できない感じになってしまっている。

nilかどうかの判別とかしないといけないから完璧なコードというわけではないけれど、

`var goshuinListData: Results<GoshuinInfo>!`って作ってあげて、そこに値を代入。`goshuinListData = realm.objects(GoshuinInfo.self)`

その後、使うところでは、`goshuinListData[indexpath.row].adress`とかやってあげると使えるよ！


全部、一旦画像の方のエラーの一個前のrealmからちゃんとデータをとるところに注力して書いてる。

<img width="479" alt="image" src="https://user-images.githubusercontent.com/49332380/226157366-aec4ad36-a243-4ac0-9243-d6c28a738f67.png">
<img width="799" alt="スクリーンショット 2023-03-19 15 21 43" src="https://user-images.githubusercontent.com/49332380/226157403-7d938f94-95de-4980-9df0-e36567844b75.png">

**Comment by txxxxc** [comment]
>tableDataのところ、私もgoshuinListだと思ってやってみたのですが、このようにエラーが出てしまいます、、


こちらの質問の回答と一緒です！！
同じ感じでResults方からArray型に変換してあげてください！！
https://github.com/lit-iphone-course/lit-iphone-course/discussions/150#discussioncomment-5352312
